scripts/_common.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+URL_REGEX_VALID='(https?|ftp|file)://[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]'
+URL_REGEX_SECURE='^(http://(127\.[0-9]+\.[0-9]+\.[0-9]+|localhost)|https://.*)(:[0-9]+)?(/.*)?$'
+
+_validate_redirect_uri() {
+    if [[ ! $target =~ $URL_REGEX_VALID ]]; then
+        ynh_die --message="Invalid destination: $target" 1
+    fi
+
+    # Avoid uncrypted remote destination with reverse proxy mode
+    # Indeed the SSO send the password in all requests in HTTP headers
+    if [[ "$redirect_type" = "reverseproxy" ]] && [[ ! $target =~ $URL_REGEX_SECURE ]]; then
+        ynh_die --message="For secure reason, you can't use an unencrypted http remote destination couple with ssowat for your reverse proxy: $target" 1
+    fi
+}
+
+_add_dummy_service() {
```

Comment:
IMHO it would be best to install another app (my_webapp) in the preinstall in tests.toml.

Reply:
Hmmyeah but my_webapp ain't really a process listening to a specific port so eh .. but yeah we could try to find some super-simple app that does run a systemd service